ambry-server/src/test/java/com/github/ambry/server/AmbryServerRequestsTest.java

```diff
@@ -939,6 +941,68 @@ public void crossColoMetricsUpdateTest() throws Exception {
     responseList.forEach(Response::release);
   }
 
+  /**
+   * Tests for appropriate responses from healthCheck admin endpoint
+   * @throws InterruptedException
+   * @throws IOException
+   */
+  @Test
+  public void healthCheckTest() throws InterruptedException, IOException {
+
+    //retrieves writeable partitions to test with
+    List<? extends PartitionId> partitionIds = clusterMap.getWritablePartitionIds(DEFAULT_PARTITION_CLASS);
+
+    for (PartitionId id : partitionIds) {
+
+      int correlationId = TestUtils.RANDOM.nextInt();
+      String clientId = TestUtils.getRandomString(10);
+
+      // Test 1: "Good" Response Excepted where the host state is in either Standby or Leader
+      HashMap<ReplicaId,ReplicaState> replicaToOriginalState = new HashMap<>(); //remembers original state of replica
+
+      //change all replica states in partition to Standby or Leader
+      for(ReplicaId replica : id.getReplicaIds()) {
+        replicaToOriginalState.put(replica, storageManager.getStore(replica.getPartitionId()).getCurrentState());
+        storageManager.getStore(replica.getPartitionId()).setCurrentState(
+            TestUtils.RANDOM.nextInt() % 2 == 0 ? ReplicaState.STANDBY : ReplicaState.STANDBY);
+      }
+
+      AdminRequest adminRequest = new AdminRequest(AdminRequestOrResponseType.HealthCheck, id, correlationId, clientId);
+      AdminResponseWithContent response =
+          (AdminResponseWithContent) sendRequestGetResponse(adminRequest, ServerErrorCode.No_Error);
+
+      assertTrue("Response not of type AdminResponse", response instanceof AdminResponse);
+      assertEquals("Response's content length does not match the content length in response",
+          response.getContent()[7],
+          response.getContent().length);
+      assertEquals("Payload was expected to be {\"health\":\"GOOD\"}","{\"health\":\"GOOD\"}",
+          new String(Arrays.copyOfRange(response.getContent(), 45, 62), StandardCharsets.UTF_8));
```

Comment:
let's fix the content generated in the handleHealthCheck method.

Reply:
Fixed